### PR TITLE
[sui-core] Deduplicate pending execution before execution

### PR DESCRIPTION
**Pull request by gdanezis**
In this PR we deal with some of the intense pending execution behaviour:
* We de-duplicate the digests to be executed and delete from the store the duplicates.
* We set a min delay of 1000ms between pending executions.

### CODE REVIEW

File: crates/sui-core/src/authority_active/execution_driver/mod.rs [hunk 1]
```diff
@@ -59,6 +59,7 @@ where
         //       transactions awaiting execution, or less often than once per transactions.
         //       However, we need to be sure that if there is an awaiting trasnactions we
         //       will eventually fire the notification and wake up here.
+        tokio::time::sleep(Duration::from_millis(1_000)).await;
```
Comment (lxfind) [inline crates/sui-core/src/authority_active/execution_driver/mod.rs:62]: Wonder if 1s too long? In https://github.com/MystenLabs/sui/pull/3718 I tried to distinguish the two cases between where some tx failed and all succeeded: one delays 500ms while the other delays only 100ms

**Review by lxfind — APPROVED** [review]

File: crates/sui-core/src/authority_active/execution_driver/mod.rs (hunk 1 above)
Comment (gdanezis) [reply crates/sui-core/src/authority_active/execution_driver/mod.rs:62]: Given that this should hardly ever happen, I am happy with the 1s for the moment, and we see

Comment (lxfind) [reply crates/sui-core/src/authority_active/execution_driver/mod.rs:62]: Well, but in the happy path (where tx execute successfully), you wouldn't want a 1s delay

Comment (lxfind) [reply crates/sui-core/src/authority_active/execution_driver/mod.rs:62]: I recommend you merge the dedup change, and leave the timing control to #3718

Comment (gdanezis) [reply crates/sui-core/src/authority_active/execution_driver/mod.rs:62]: Ah sorry did not see this in time -- feel free to refactor and remove.

**Pull request merged**